src/commonMain/kotlin/org/kbson/BSONException.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kbson
+
+/** A general runtime exception raised in BSON processing. */
+open class BSONException(
+    message: String? = null,
+    cause: Throwable? = null,
+    val errorCode: Int? = null
+) : RuntimeException(message, cause) {
```

Comment:
Maybe add a custom `toString()` method in this class that include the errorCode in the output?